### PR TITLE
Slack interface

### PR DESCRIPTION
**Pull request by ejholmes**
This is a little spike into a Slack interface to Empire for https://github.com/remind101/empire/issues/686. **Not ready to merge**

The way I built it, there's now a `cli` package that provides a CLI to Empire (whether it's in process or out of process doesn't matter) and there's just a slash handler that serves the CLI to Slack.

**Things still todo**

* [x] Wrap slack server with [slash.ValidateToken](https://godoc.org/github.com/ejholmes/slash#ValidateToken)
* [ ] Authenticate slack users against github. (Currently just runs as the `slack` user).
* [ ] Docs
* [ ] Add ability to disable certain commands.

### CODE REVIEW

File: cli/scale.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"errors"
+	"strconv"
+	"strings"
+)
+
+var errInvalidScaleArg = errors.New("invalid argument")
+
+// This is a direct copy paste from
+// https://github.com/remind101/emp/blob/6fc34ef0633f5a26217023ede86b866bf4f17cc4/scale.go#L127-L168.
+//
```
Comment (sanjayprabhu) [inline cli/scale.go]: I assume it's not worth sharing any code between this and `emp`?

**Comment by sanjayprabhu** [comment]
Nice, this is looking great!

File: cli/scale.go (hunk 1 above)
Comment (ejholmes) [inline cli/scale.go]: That's definitely the end goal. If https://github.com/remind101/empire/pull/712 gets merged then this can be shared.

**Comment by phobologic** [comment]
This just came up internally, but we should probably have a way for people to disable certain commands.  Here, for example, we'd definitely not want to allow `run` over slack, since that goes against some of our security principles.  `scale` is questionably on that spectrum as well.

**Comment by ejholmes** [comment]
Added to the TODO section.

**Comment by ejholmes** [comment]
This probably won't get any love anytime soon, so gonna close for now.

**Pull request closed without merging**